src/modules/azure-azcopy-sync.test.ts

```diff
@@ -0,0 +1,477 @@
+
+            // Act
+            AzureAzcopySync.localFile(localFileOptions);
+
+            // Assert
+            expect(spawnSyncSpy).toHaveBeenCalledWith(
+                command,
+                calculatedCommandArgs,
+                spawnSyncArgs
+            );
+        });
+
+        test("when destination.account provided then used in the destination URL argument", () => {
+            // Arrange
+            localFileOptions.destination.account = "other_test_account";
+            calculatedCommandArgs[2] = _getBlobStorageUrl(
+                localFileOptions.destination
+            );
+
+            // Act
+            AzureAzcopySync.localFile(localFileOptions);
+
+            // Assert
+            expect(spawnSyncSpy).toHaveBeenCalledWith(
+                command,
+                calculatedCommandArgs,
+                spawnSyncArgs
+            );
+        });
+
+        test("when destination.container provided then used in the destination URL argument", () => {
+            // Arrange
+            localFileOptions.destination.container = "other_test_container";
+            calculatedCommandArgs[2] = _getBlobStorageUrl(
+                localFileOptions.destination
+            );
+
+            // Act
+            AzureAzcopySync.localFile(localFileOptions);
+
+            // Assert
+            expect(spawnSyncSpy).toHaveBeenCalledWith(
+                command,
+                calculatedCommandArgs,
+                spawnSyncArgs
+            );
+        });
+
+        test("when destination.path provided then used in the destination URL argument", () => {
+            // Arrange
+            localFileOptions.destination.path = "other_test_account";
+            calculatedCommandArgs[2] = _getBlobStorageUrl(
+                localFileOptions.destination
+            );
+
+            // Act
+            AzureAzcopySync.localFile(localFileOptions);
+
+            // Assert
+            expect(spawnSyncSpy).toHaveBeenCalledWith(
+                command,
+                calculatedCommandArgs,
+                spawnSyncArgs
+            );
+        });
+
+        test("when destination.path not provided then not used in the destination URL argument", () => {
+            // Arrange
+            delete localFileOptions.destination.path;
+            calculatedCommandArgs[2] = _getBlobStorageUrl(
+                localFileOptions.destination
+            );
+
+            // Act
+            AzureAzcopySync.localFile(localFileOptions);
+
+            // Assert
+            expect(spawnSyncSpy).toHaveBeenCalledWith(
+                command,
+                calculatedCommandArgs,
+                spawnSyncArgs
+            );
+        });
+
+        test("when destination.sasToken provided then used in the destination URL argument", () => {
+            // Arrange
+            localFileOptions.destination.sasToken = "other_test_sas_token";
+            calculatedCommandArgs[2] = _getBlobStorageUrl(
```

Comment:
Not sure I'm a huge fan of pre-building the expected arg array each time - indexing into it for modification left me wondering what the original value was and why we're changing it.

Reply:
The only other way I could think to set this up was to create a function per set of expected command args (one for localFile() and one for containers().  Each function would then need to call the appropriate function then update it's returned value like it is here.  I'll make that change here so it's a little more clear.